### PR TITLE
Fix pull-cluster-api-vendor-in-sync presubmit

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
none
```

/assign @vincepri

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@vincepri I've removed the final `git reset --hard` because it may surprise ppl, PTAL

File: scripts/ci-is-vendor-in-sync.sh [hunk 1]
```diff
@@ -20,4 +20,6 @@ set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 
-cd $REPO_ROOT && dep status
+cd $REPO_ROOT
```
Comment (roberthbailey) [inline scripts/ci-is-vendor-in-sync.sh]: Should this use a make target, e.g. the one added in https://github.com/kubernetes-sigs/cluster-api/pull/692?

Comment (alvaroaleman) [reply scripts/ci-is-vendor-in-sync.sh]: No, this should check if vendor is in sync not bring vendor in sync. The make target in #692 does the latter

**Pull request merged**